src/extension/byok/vscode-node/azureProvider.ts

```diff
@@ -65,4 +70,87 @@ export class AzureBYOKModelProvider extends CustomOAIBYOKModelProvider {
 	protected override resolveUrl(modelId: string, url: string): string {
 		return resolveAzureUrl(modelId, url);
 	}
+
+	protected override async getModelsWithCredentials(silent: boolean): Promise<BYOKKnownModels> {
+		// Check user's authentication preference from settings github.copilot.chat.azureAuthType (default: AzureAuthMode.EntraId)
+		const authType = this._configurationService.getConfig(ConfigKey.AzureAuthType);
+
+		if (authType === AzureAuthMode.EntraId) {
+			// Pre-authenticate during model enumeration (not when sending message)
+			// This mirrors API key behavior where user is prompted during enumeration
+			if (!silent) {
+				try {
+					await vscode.authentication.getSession(
+						AzureAuthMode.MICROSOFT_AUTH_PROVIDER,
+						[AzureAuthMode.COGNITIVE_SERVICES_SCOPE],
+						{ createIfNone: true }
+					);
+				} catch (error) {
+					// If sign-in fails, don't show models in picker
+					this._logService.error('[AzureBYOKModelProvider] Authentication failed during Entra ID sign-in:', error);
+					return {};
+				}
+			}
+
+			// Return all configured models (no API key check needed for Entra ID)
+			return this.getAllModels();
+		} else {
+			// API KEY MODE: Use traditional API key authentication
+			return super.getModelsWithCredentials(silent);
+		}
+	}
+
+	override async provideLanguageModelChatResponse(
+		model: CustomOAIModelInfo,
+		messages: Array<LanguageModelChatMessage | LanguageModelChatMessage2>,
+		options: ProvideLanguageModelChatResponseOptions,
+		progress: Progress<LanguageModelResponsePart2>,
+		token: CancellationToken
+	): Promise<void> {
+		const authType = this._configurationService.getConfig(ConfigKey.AzureAuthType);
+
+		if (authType === AzureAuthMode.EntraId) {
+			const session = await vscode.authentication.getSession(
+				AzureAuthMode.MICROSOFT_AUTH_PROVIDER,
+				[AzureAuthMode.COGNITIVE_SERVICES_SCOPE],
+				{ createIfNone: true }
```

Comment:
`session` is always defined when you do `createIfNone: true`. The types should say `session` is an AuthSession.